src/WooCommerce.php

```diff
@@ -128,6 +128,148 @@ public static function woocommerce_single_product_image_thumbnail_html($html, $t
     return preg_replace('/(<a\s+)/i', '<a data-srcset="' . $srcset . '" data-sizes="' . $srcsizes . '" ', $html);
   }
 
+  /**
+   * Adds additional images from product variation gallery.
+   *
+   * @implements woocommerce_product_get_gallery_image_ids
+   */
+  public static function woocommerce_product_get_gallery_image_ids($value, $obj) {
+    global $product;
+
+    if (!$product || $product->get_type() !== 'variable') {
+      return $value;
+    }
+
+    $variations = $product->get_available_variations();
```

Comment:
This should use the new argument for performance.

But please also consult with @bogdanarizancu to check whether a custom query would be more appropriate for performance for this use-case, as it seems we only need the images.  (I think I've seen code for that already somewhere)